src/ceph-disk

```diff
@@ -527,6 +548,70 @@ def get_partition_dev(dev, pnum):
     else:
         raise Error('partition %d for %s does not appear to exist' % (pnum, dev))
 
+def get_blkid_udev(dev,output_set=BLKID_UDEV):
+    """
+    Present a consistent list of blkid_udev what ever the device.
+    Paramaeters output_set is the keys in the dictionary of
+    posible outputs.
+
+    defaults to a complete list of output.
+
+    Since  "blkid -p -o udev", returns output the is depenet on type
+    We need some code to make this command always succeed.
+    the output_set defines the list of paramters.
+    """
+    def mdl_blkid_udev_init(local_model):
+        mdl_keys = set(local_model.keys())
+        missing = output_set.difference(mdl_keys)
+        for item in missing:
+            local_model[item] = None
+
+    def blkid_udev_read(local_mode,blkid_udev_output):
+        for line in blkid_udev_output.splitlines():
+            (key, value) = line.split('=')
+            if key in output_set:
+                local_mode[key] = value
+
+    model = {}
+    mdl_blkid_udev_init(model)
+    blkid, _ = command(
```

Comment:
In some cases blkid does not work https://github.com/dachary/ceph/commit/9e796dc8b29c7f1a44d2aebcd0f59fa915270822. It can be repeated every time but I don't have a clue as to why it really happens. Using sgdisk works cross platform and reliably.